.changeset/plenty-pandas-smoke.md

```diff
@@ -0,0 +1,17 @@
+---
+'@keystonejs/keystone': major
+---
+
+Introduced `ListAuthProvider` to generate authentication mutations and queries. This introduces the following breaking changes:
+ * `Keystone.createAuthStrategy()` must be called *after* the associated `List` has been created.
+ * `List.hasAuth()` has been removed.
+ * `List.getAuth()` has been removed.
+ * `List.gqlNames.authenticatedQueryName` has been removed.
+ * `List.gqlNames.authenticateMutationPrefix` has been removed.
+ * `List.gqlNames.unauthenticateMutationName` has been removed.
+ * `List.gqlNames.authenticateOutputName` has been removed.
+ * `List.gqlNames.unauthenticateOutputName` has been removed.
+ * `List.checkAuthAccess()` has been removed.
+ * `List.authenticatedQuery()` has been removed.
+ * `List.authenticateMutation()` has been removed.
+ * `List.unauthenticateMutation()` has been removed.
```

Comment:
This looks significant and scary. Although I doubt the impact outside this team is large can we add a note about how you migrate code that uses these methods?

Reply:
These methods can now be found on the `AuthListProvider` object, however they generally should not be considered part of the Keystone public API. Users who wish to access them can still find them in `keystone._providers`.